---
title: "Clustering Pseudotime Inference on Skin Cell scRNA-seq"
author: "Matt Karikomi"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Clustering Pseudotime Inference on Skin Cell scRNA-seq}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
rm(list = ls())
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
set.seed(13)
```

This is an example of clustering, marker identification and pseudotime inference on a set of skin cells.

## Load and Preprocess Data

Load and log transform the data

```{r, results='asis'}
library(RSoptSC)
data("GSE67602_Joost")
logdata <- log10(GSE67602_Joost$data + 1)
```

### Remove Spike-in RNA
```{r}
# clean the gene names
gene_names <- GSE67602_Joost$gene_names
spikein <- grep('ERCC', gene_names)
gene_names <- gene_names[-spikein]

# clean the data
logdata <- logdata[-spikein,]
```

Apply number of features and exclusion threshold

```{r, results='asis'}
gene_expression_threshold <- 0.03
n_features <- 3000
filtered_data <- SelectData(logdata, gene_expression_threshold, n_features)
```

## Compute the Similarity Matrix

Run L2R2 on the processed data, outputting the number of iterations and value of the objective.

```{r, results='asis'}
sim <- SimilarityM(lambda = 0.05, data = filtered_data$M_variable)
```

Compute distances on the similarity matrix.  If the truncated graph has multiple components, join them to improve inter-cluster distance estimation.

```{r, results='asis'}
mapped <- RepresentationMap(similarity_matrix = sim$W,
                            join_components = TRUE)
```

## Cluster the Cells

Infer cluster number using spectra of the ensemble Laplacian.

```{r, results='asis'}
clust <- CountClusters(data = sim$W)
n_clusters <- clust$n
```

In this analysis, we obtain bounds on the number of clusters, where the lower bound is equal to the number of zero eigenvalues and the upper bound is the index of the eigenvalue preceding the largest eigengap.  We take this value as the cluster number here.

```{r, results='asis'}
plot(c(1:20), 
     clust$eigs$val[1:20],
     xlab = NA,
     ylab = 'eigenvalues',
     main = 'Eigenvalues of the Graph Laplacian')
```


Factor the similarity matrix such that W = HxH', using the computed cluster number for the rank of H.

```{r, results='asis', warning=FALSE}
res <- NMF::nmf(x = sim$W,
              rank = n_clusters,
              method = 'lee',
              seed = 'nndsvd',
              .options = 'nP');
H <- NMF::basis(res)
```

Assign cluster labels to the cells.

```{r, results='asis'}
labels <- apply(H, 1, function(x){
  which(x == max(x))})
```


### Visualize the Clusters
Cells are labeled by cluster membership and plotted on low-dimensional embedding (t-SNE) of the similarity matrix.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
# plot clusters
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                         feature = as.factor(labels),
                         title = "NMF Cluster Labeling",
                         subtitle = "t-SNE Embedding",
                         featurename = "Cluster ID")
```

As described in Joost et al (2016), Subpopulations including Basal, IFE-DI, IFE-DII, IFE-KI, IFE-KII were identified by standard techniques.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
true_labels <- RSoptSC::GSE67602_Joost$annotation
# plot clusters
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                         feature = as.factor(true_labels),
                         title = "True Labeling",
                         subtitle = "t-SNE Embedding",
                         featurename = "Annotated Cell Types")
```

## Find cluster marker genes.

Previously, we clustered the cells using L2R2/Sym-NMF, now we analyze the markers of the clusters.  We look at the normalized expression values of all genes using pre-selection parameters similar to the pre-clustering selection parameters.  The top six markers selected as described in Wang et. al.  As expected, markers Krt14, Krt10, and Lor are inferred for clusters overlapping with IHC-based Basal, Differentiated (IFE-DI, IFE-DII), and Keratinized (IFE-DI, IFE-DII) cell annotations.   

```{r, results='asis'}
library(dplyr)
n <- 6
n_cells <- length(labels)


markers <- GetMarkerTable(counts_data = GSE67602_Joost$data,
                          cluster_labels = labels,
                          H = H,
                          gene_expression_threshold = 6,
                          n_features = 20000)

sorted_cell <- sort.int(labels, index.return = TRUE)

markers_table <- as.data.frame(markers)
sortedmarkers <- dplyr::arrange(markers_table, clusterId, desc(geneScore))
sorted_gene_table <- tibble::as_tibble(sortedmarkers) %>% group_by(clusterId) %>% top_n(n, geneScore)
sorted_gene <- sorted_gene_table$geneID

plot_data <- GSE67602_Joost$data[sorted_gene, sorted_cell$ix]
# plot_data[plot_data == 0] <- 1
# plot_data <- scale(plot_data)

rlabs <- GSE67602_Joost$gene_names[sorted_gene]

clabs <- c(rep(NA, n_cells))
counts <- as.matrix(table(labels))
accumulator <- matrix(1, n_clusters, n_clusters)*lower.tri(matrix(1, n_clusters, n_clusters), diag = TRUE)
last_cells <- accumulator %*% counts
offset <- floor(counts/2)
labeled_columns <- last_cells - offset
clabs[labeled_columns] <- c(1:n_clusters)

p <- gplots::heatmap.2(log10(plot_data+1),
                  col = rev(brewer.pal(11,"RdBu")),
                  trace = 'none',
                  dendrogram='none',
                  Rowv=FALSE, Colv=FALSE,
                  labCol = clabs,
                  labRow = rlabs,
                  srtCol = 0,
                  cexCol =1,
                  cexRow = .7,
                  lhei = c(1,3.5),
                  key.title = NA,
                  scale = "row")

```

## Pseudotime Inference

Compute cluster-cluster graph and find the root cluster.

```{r, results='asis'}
cluster_ptime <- FindRootCluster(cluster_labels = labels,
                                 flat_embedding = mapped$flat_embedding,
                                 dist_graph = mapped$dist_graph,
                                 dist_flat = mapped$dist_flat,
                                 reverse = TRUE)
```

Compute the cell-cell graph and find the root cell.

```{r, results='asis'}
root_cell <- FindRootCell(use_flat_dist = FALSE,
                          cluster_order_by = "distance",
                          cell_order_by = "distance",
                          graph_cluster_mst = cluster_ptime$cluster_mst,
                          dist_graph  = mapped$dist_graph,
                          dist_flat = mapped$dist_flat,
                          cluster_labels = labels,
                          root_cluster = cluster_ptime$root_cluster)
```

## Pseudotime Visualization

Compute the directional minimum spanning tree on the cluster-cluster graph for visualization.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
library(ggnetwork)
cluster_predecessors <- GetPredecessors(cluster_ptime$cluster_mst, cluster_ptime$root_cluster)
cluster_dtree <- GetDominatorTree(cluster_predecessors, cluster_ptime$graph_cluster)
PlotLineage(cluster_dtree)
```


Compute the path-length vector from root, and plot the cells in the previously computed low-dimensional embedding.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
pseudotime <- mapped$dist_graph[root_cell,]

# plot pseudotime
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                         feature = pseudotime,
                         title = "Pseudotime Labeling",
                         subtitle = "t-SNE Embedding",
                         featurename = "Relative L2R2 Distance from Root")
```

## Plot Gene Expression on the Flat Embedding

Plot Expression of Krt14, Krt10, and Lor in t-SNE Space:

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
gene_index <- which(gene_names == 'Krt14')
labeling <- logdata[gene_index,]

# plot features
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                    feature = labeling,
                    title = "Krt14 Expression",
                    subtitle = "t-SNE Embedding",
                    featurename = "Log Krt14 Expression")
```

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
gene_index <- which(gene_names == 'Krt10')
labeling <- logdata[gene_index,]

# plot features
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                    feature = labeling,
                    title = "Krt10 Expression",
                    subtitle = "t-SNE Embedding",
                    featurename = "Log Krt10 Expression")
```

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
gene_index <- which(gene_names == 'Lor')
labeling <- logdata[gene_index,]

# plot features
FeatureScatterPlot(flat_embedding = mapped$flat_embedding,
                    feature = labeling,
                    title = "Lor Expression",
                    subtitle = "t-SNE Embedding",
                    featurename = "Log Lor Expression")
```

## Plot Gene Expression Levels in Cells Grouped by Cluster

Violin Plot of Krt14 expression.  Note strong expression of the basal compartment (clusters 1 and 6):

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
  ViolinPlotExpression(data = logdata,
                       gene_names = gene_names,
                       labels = labels,
                       gene_name = "Krt14")
```
Violin Plot of Krt10 expression.  Note strong expression in the IFE-DI/DII compartments (clusters 2, 4, 5, 7):

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
  ViolinPlotExpression(data = logdata,
                       gene_names = gene_names,
                       labels = labels,
                       gene_name = "Krt10")
```
Violin Plot of Lor expression.  Note strong expression in cluster 3, which completely encompasses IFE-KI/KII compartments:

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
  ViolinPlotExpression(data = logdata,
                       gene_names = gene_names,
                       labels = labels,
                       gene_name = "Lor")
```

## Analyze Signaling Mediated by a Ligand-Receptor Family

We present a family of ligan-receptor pairs along with their up/down-regulated targets.  We compute the probability of intercellular interaction based on this pathway.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
  ligands <- list(Tgfb1 = list(Tgfbr1 = list(up = list('Zeb2','Smad2','Wnt4','Wnt11','Bmp7','Sox9','Notch1'),
                                             down = list('Crebbp','Fos','Id1','Jun','Runx1','Smad1','Smad5','Sox4','Cdh1')),
                               Tgfbr2 = list(up = list('Zeb2','Smad2','Wnt4','Wnt11','Bmp7','Sox9','Notch1'),
                                             down = list('Crebbp','Fos','Id1','Jun','Runx1','Smad1','Smad5','Sox4','Cdh1'))),
                  Tgfb2 = list(Tgfbr1 = list(up = list('Zeb2','Smad2','Wnt4','Wnt11','Bmp7','Sox9','Notch1'),
                                             down = list('Crebbp','Fos','Id1','Jun','Runx1','Smad1','Smad5','Sox4','Cdh1')),
                               Tgfbr2 = list(up = list('Zeb2','Smad2','Wnt4','Wnt11','Bmp7','Sox9','Notch1'),
                                             down = list('Crebbp','Fos','Id1','Jun','Runx1','Smad1','Smad5','Sox4','Cdh1'))))
  Pmats <- GetSignalingPartners(logdata,
                                  gene_names,
                                  ligands)
```

### Circos Plot of Intercellular Network
Now plot the previous result as a circos plot on the cells in the dataset, ordered by cluster.

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7}
  sigonly = FALSE   # If true => input is only signaler cells, and P will not be thresholded
  subsample = TRUE
  threshold = 0.05
  nsample = 100

  plotSig(P = Pmats$P_agg,
          cluster_label = labels,
          threshold = threshold,
          sigonly = sigonly,
          subsample = subsample,
          nsample = nsample,
          plottitle = "Network Inference on the Activation and Suppression of Tgfb Targets ")
```
Visualize the expression of ligand/receptor and up/down target expression across clusters

```{r, results='asis', out.width = '100%', out.height = '100%', fig.width = 7, fig.heigt = 7, eval=FALSE}
  sigonly = FALSE   # If true => input is only signaler cells, and P will not be thresholded
  subsample = FALSE
  threshold = 0.05

  Pc <- ClusterSig(P, sorted_cell$ix, sorted_cell$x)
  
  plotSig(Pc = Pmats$P_agg,
          cluster_label = labels,
          threshold = threshold,
          sigonly = sigonly,
          subsample = subsample,
          plottitle = "Network Inference on the Activation and Suppression of Tgfb Targets ")
```


### Accuracy of SoptSC Clustering
Use adjusted rand index (ARI) and normalized mutual information (NMI) to assess performance:

```{r, echo=FALSE, results='asis'}
library(knitr)
true_labels <- as.numeric(as.factor(RSoptSC::GSE67602_Joost$annotation))
matlab_labels <- as.numeric(RSoptSC:::JoostMarkers$cluster_label)

ARIresClusterMatlab <- ClusterR::external_validation(true_labels = true_labels,
                                        clusters = matlab_labels,
                                        method = 'adjusted_rand_index')
ARIresCluster <- ClusterR::external_validation(true_labels = true_labels,
                                  clusters = as.numeric(labels),
                                  method = 'adjusted_rand_index')
NMIresClusterMatlab <- ClusterR::external_validation(true_labels = true_labels,
                                           clusters = matlab_labels,
                                           method = 'nmi')
NMIresCluster <- ClusterR::external_validation(true_labels = true_labels,
                                     clusters = as.numeric(labels),
                                     method = 'nmi')

cluster_table <- data.frame(ARI = c(ARIresCluster, ARIresClusterMatlab), 
           NMI = c(NMIresCluster, NMIresClusterMatlab),
           row.names = c('RSoptSC', 'SoptSC'))

kable(cluster_table, caption = 'Performance Comparison Between RSoptSC and Popular Methods')
```
