<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Import PBMC 10X Data (Cell Ranger 3.1.0 Output) â€¢ RSoptSC</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Import PBMC 10X Data (Cell Ranger 3.1.0 Output)">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RSoptSC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RSoptSC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/PBMC_10X.html">Import PBMC 10X Data (Cell Ranger 3.1.0 Output)</a>
    </li>
    <li>
      <a href="../articles/QC_SSC.html">Direct Download from GEO with Annotation for Skin Cell scRNA-seq</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mkarikom/RSoptSC">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Import PBMC 10X Data (Cell Ranger 3.1.0 Output)</h1>
                        <h4 class="author">Matt Karikomi</h4>
            
            <h4 class="date">2019-09-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mkarikom/RSoptSC/blob/master/vignettes/PBMC_10X.Rmd"><code>vignettes/PBMC_10X.Rmd</code></a></small>
      <div class="hidden name"><code>PBMC_10X.Rmd</code></div>

    </div>

    
    
<p>This is an example of clustering, marker identification and pseudotime inference on a set of peripheral blood mononuclear cells (PBMCs).</p>
<div id="load-and-preprocess-data-download-from-10x-genomics" class="section level2">
<h2 class="hasAnchor">
<a href="#load-and-preprocess-data-download-from-10x-genomics" class="anchor"></a>Load and Preprocess Data (Download from 10X Genomics)</h2>
<p>RSoptSC::ReadFilter10X is a wrapper for <a href="https://www.google.com">scran</a>, <a href="https://www.google.com">scater</a>, <a href="https://www.google.com">DropletUtils</a>, whose documentation is a must-read for experimentalists.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">url =<span class="st"> "http://cf.10xgenomics.com/samples/cell-exp/3.1.0/5k_pbmc_protein_v3_nextgem/5k_pbmc_protein_v3_nextgem_filtered_feature_bc_matrix.tar.gz"</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">destfile =<span class="st"> '/tmp/counts.tar.gz'</span> <span class="co"># all data contained in "filtered_feature_bc_matrix"</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">extdir =<span class="st"> '/tmp'</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"></a>
<a class="sourceLine" id="cb1-6" data-line-number="6">f =<span class="st"> </span><span class="kw">CFILE</span>(destfile, <span class="dt">mode=</span><span class="st">"wb"</span>)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">curlPerform</span>(<span class="dt">url =</span> url, <span class="dt">writedata =</span> f<span class="op">@</span>ref)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/base/connections.html">close</a></span>(f)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10">utils<span class="op">::</span><span class="kw"><a href="https://rdrr.io/r/utils/untar.html">untar</a></span>(<span class="dt">tarfile =</span> destfile, <span class="dt">verbose =</span> <span class="ot">TRUE</span>, <span class="dt">exdir =</span> extdir)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"></a>
<a class="sourceLine" id="cb1-12" data-line-number="12">upper &lt;-<span class="st"> </span><span class="dv">3000</span> <span class="co"># number of variable genes</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">cache &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ReadFilter10X.html">ReadFilter10X</a></span>(<span class="dt">filepath =</span> <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(extdir, <span class="st">"/filtered_feature_bc_matrix"</span>), </a>
<a class="sourceLine" id="cb1-14" data-line-number="14">                       <span class="dt">dataformat =</span> <span class="st">'counts'</span>, </a>
<a class="sourceLine" id="cb1-15" data-line-number="15">                       <span class="dt">feature_annotation_key =</span> <span class="st">"ENSEMBL"</span>,</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">                       <span class="dt">annotation_db =</span> <span class="st">'org.Hs.eg.db'</span>,</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">                       <span class="dt">pre_clust =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">                       <span class="dt">rescale =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb1-19" data-line-number="19">                       <span class="dt">filtercells =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb1-20" data-line-number="20">                       <span class="dt">subsample =</span> <span class="dv">700</span>)</a></code></pre></div>
<p><img src="PBMC_10X_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">all_data &lt;-<span class="st"> </span>SingleCellExperiment<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SCE-assays.html">logcounts</a></span>(cache<span class="op">$</span>qc)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">similarity_data &lt;-<span class="st"> </span>SingleCellExperiment<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SCE-assays.html">logcounts</a></span>(cache<span class="op">$</span>qc[cache<span class="op">$</span>vargenes<span class="op">$</span>ind[<span class="dv">1</span><span class="op">:</span>upper],])</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">gene_names &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(similarity_data)</a></code></pre></div>
</div>
<div id="compute-the-similarity-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-the-similarity-matrix" class="anchor"></a>Compute the Similarity Matrix</h2>
<p>Run L2R2 on the processed data, outputting the number of iterations and value of the objective.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">S &lt;-<span class="st"> </span><span class="kw"><a href="../reference/SimilarityM.html">SimilarityM</a></span>(<span class="dt">lambda =</span> <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                 <span class="dt">data =</span> similarity_data,</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                 <span class="dt">dims =</span> <span class="dv">3</span>,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                 <span class="dt">pre_embed_method =</span> <span class="st">'tsne'</span>,</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                 <span class="dt">perplexity =</span> <span class="dv">20</span>, </a>
<a class="sourceLine" id="cb3-6" data-line-number="6">                 <span class="dt">pca_center =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb3-7" data-line-number="7">                 <span class="dt">pca_scale =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Compute distances on the similarity matrix. If the truncated graph has multiple components, join them to allow inter-cluster distance estimation.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">low_dim_mapping &lt;-<span class="st"> </span><span class="kw"><a href="../reference/RepresentationMap.html">RepresentationMap</a></span>(<span class="dt">similarity_matrix =</span> S<span class="op">$</span>W,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                                      <span class="dt">flat_embedding_method =</span> <span class="st">'tsne'</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                                      <span class="dt">join_components =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                                      <span class="dt">perplexity =</span> <span class="dv">35</span>,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                                      <span class="dt">theta =</span> <span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                                      <span class="dt">normalize =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">                                      <span class="dt">pca =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">                                      <span class="dt">pca_center =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">                                      <span class="dt">pca_scale =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">                                      <span class="dt">dims =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">                                      <span class="dt">initial_dims =</span> <span class="dv">2</span>)</a></code></pre></div>
</div>
<div id="cluster-the-cells" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-the-cells" class="anchor"></a>Cluster the Cells</h2>
<p>Cluster the cells by factoring the similarity matrix. Infer cluster number using spectra of the ensemble Laplacian. In this analysis, we obtain bounds on the number of clusters, where the lower bound is equal to the number of zero eigenvalues and the upper bound is the index of the eigenvalue preceding the largest eigengap. We take this value as the cluster number.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">clusters &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ClusterCells.html">ClusterCells</a></span>(<span class="dt">similarityMatrix =</span> S<span class="op">$</span>W, <span class="dt">n_comp =</span> <span class="dv">15</span>, <span class="dt">.options=</span><span class="st">'p'</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">H &lt;-<span class="st"> </span>clusters<span class="op">$</span>H</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">labels &lt;-<span class="st"> </span>clusters<span class="op">$</span>labels</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">n_clusters &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(clusters<span class="op">$</span>labels))</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>), </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">     clusters<span class="op">$</span>ensemble<span class="op">$</span>eigs<span class="op">$</span>val[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>],</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">     <span class="dt">xlab =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">     <span class="dt">ylab =</span> <span class="st">'eigenvalues'</span>,</a>
<a class="sourceLine" id="cb5-10" data-line-number="10">     <span class="dt">main =</span> <span class="st">'Eigenvalues of the Graph Laplacian'</span>)</a></code></pre></div>
<p><img src="PBMC_10X_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div id="visualize-the-clusters" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-the-clusters" class="anchor"></a>Visualize the Clusters</h3>
<p>Cells are labeled by cluster membership and plotted on low-dimensional embedding (t-SNE) of the similarity matrix.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># define a scheme for the coloring.  This is a required parameter for plotting discrete (factor) data</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">colorscale &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ColorHue.html">ColorHue</a></span>(<span class="dt">n =</span> <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(labels)))</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">colorscale &lt;-<span class="st"> </span>colorscale<span class="op">$</span>hex.<span class="fl">1.</span>n.</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co"># plot clusters</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="kw"><a href="../reference/FeatureScatterPlot.html">FeatureScatterPlot</a></span>(<span class="dt">flat_embedding =</span> low_dim_mapping<span class="op">$</span>flat_embedding,</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">                   <span class="dt">feature =</span> <span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(labels),</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">                   <span class="dt">title =</span> <span class="st">"NMF Cluster Labeling"</span>,</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">                   <span class="dt">subtitle =</span> <span class="st">"t-SNE Embedding"</span>,</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">                   <span class="dt">featurename =</span> <span class="st">"Cluster ID"</span>,</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">                   <span class="dt">colorscale =</span> colorscale)</a></code></pre></div>
<p><img src="PBMC_10X_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#load-and-preprocess-data-download-from-10x-genomics">Load and Preprocess Data (Download from 10X Genomics)</a></li>
      <li><a href="#compute-the-similarity-matrix">Compute the Similarity Matrix</a></li>
      <li><a href="#cluster-the-cells">Cluster the Cells</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Matt Karikomi, Shuxiong Wang, Adam MacLean.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
