<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read Counts Data — ReadAnnotateCountsSCE • RSoptSC</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Read Counts Data — ReadAnnotateCountsSCE" />

<meta property="og:description" content="Read counts data from file and annotate the generated Single Cell Experiment Object.  Right now this function assumes:
- mouse cells
- only spike ins are ERCC
- row and column names are provided in the data
- the row names are gene symbols
In the case of GEO data, it takes a GSE accession code and assumes that the supp files contain all the expression data" />

<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RSoptSC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RSoptSC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QC_SSC.html">Direct Download from GEO with Annotation for Skin Cell scRNA-seq</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read Counts Data</h1>
    
    <div class="hidden name"><code>ReadAnnotateCountsSCE.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Read counts data from file and annotate the generated Single Cell Experiment Object.  Right now this function assumes:
- mouse cells
- only spike ins are ERCC
- row and column names are provided in the data
- the row names are gene symbols
In the case of GEO data, it takes a GSE accession code and assumes that the supp files contain all the expression data</p>
    
    </div>

    <pre class="usage"><span class='fu'>ReadAnnotateCountsSCE</span>(<span class='kw'>filepath</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>inf_exp_rate</span> <span class='kw'>=</span> <span class='fl'>0</span>,
  <span class='kw'>inf_gene_count</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>inf_cell_count</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='no'>series_accession</span>,
  <span class='kw'>metad_title</span> <span class='kw'>=</span> <span class='st'>"characteristics_ch1"</span>, <span class='kw'>cell_id</span> <span class='kw'>=</span> <span class='st'>"title"</span>,
  <span class='kw'>pheno_filter</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>plot_title</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>GEOfilecache</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/getwd'>getwd</a></span>(),
  <span class='no'>annotation_db</span>, <span class='no'>feature_annotation_key</span>, <span class='kw'>remove_spike_ERCC</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>pre_clust</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>rescale</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>var_gene_method</span> <span class='kw'>=</span> <span class='st'>"mean_var"</span>,
  <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>filepath</th>
      <td><p>the full or relative (to the workspace) path of the data file</p></td>
    </tr>
    <tr>
      <th>inf_exp_rate</th>
      <td><p>the minimum mean expression rate of retained genes</p></td>
    </tr>
    <tr>
      <th>inf_gene_count</th>
      <td><p>retain a gene if more than this number of cells express it</p></td>
    </tr>
    <tr>
      <th>inf_cell_count</th>
      <td><p>retain a cell if more than this number of genes are expressed in it</p></td>
    </tr>
    <tr>
      <th>series_accession</th>
      <td><p>the series accession code, eg "GSE67602"</p></td>
    </tr>
    <tr>
      <th>metad_title</th>
      <td><p>the series accession code, eg "GSE67602"</p></td>
    </tr>
    <tr>
      <th>cell_id</th>
      <td><p>the series accession code, eg "GSE67602"</p></td>
    </tr>
    <tr>
      <th>pheno_filter</th>
      <td><p>a vector of strings to partially match cell metadata</p></td>
    </tr>
    <tr>
      <th>plot_title</th>
      <td><p>path of the saved mean/variance plot, default NULL, if NULL the plot will be printed instead</p></td>
    </tr>
    <tr>
      <th>GEOfilecache</th>
      <td><p>location for caching series supplemental files, default is current directory</p></td>
    </tr>
    <tr>
      <th>annotation_db</th>
      <td><p>annotation package for the data, eg "org.Mm.eg.db"</p></td>
    </tr>
    <tr>
      <th>feature_annotation_key</th>
      <td><p>the key for annotation.db lookups, eg "SYMBOL"</p></td>
    </tr>
    <tr>
      <th>remove_spike_ERCC</th>
      <td><p>remove ERCC spike ins after they are used for normalization</p></td>
    </tr>
    <tr>
      <th>pre_clust</th>
      <td><p>whether or not to precluster the cells before getting size factors</p></td>
    </tr>
    <tr>
      <th>rescale</th>
      <td><p>whether to compute size factors and rescale counts</p></td>
    </tr>
    <tr>
      <th>var_gene_method</th>
      <td><p>how to select variable genes, eg 'mean_var', 'pca'.  mean_var selects the genes whose variance is highest after removing the fitted spike-in-based technical variance. pca finds the genes with the max rotation coefficients</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>arguments to called functions</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list containing:</p>
<dt>c(seur, sce)</dt><dd><p>a seurat or sce object</p></dd>

    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Matt Karikomi, Shuxiong Wang, Adam MacLean.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

